internal/infrastructure/kubernetes/proxy/resource_provider.go

```diff
@@ -193,36 +193,27 @@ func (r *ResourceRender) Deployment() (*appsv1.Deployment, error) {
 	}
 	deploymentConfig := provider.GetEnvoyProxyKubeProvider().EnvoyDeployment
 
+	// If deployment config is nil, it's not Deployment installation.
```

Comment:
we should never be reaching here, right ?

Reply:
this just protects us from a null ptr dereference below, im guessing, if so would be good to pen that down as a comment

Reply:
> we should never be reaching here, right ?

No. In envoyProxy use DaemonSet, `provider.GetEnvoyProxyKubeProvider().EnvoyDeployment` return `nil`, then `(r *ResourceRender) Deployment()` return `nil`, then in function `createOrUpdateDeployment` `deployment = nil`, then need to delete previos deployment (if exist)